### ISSUE TITLE
Fetching image using react-pdf <Image> causes CORS issue, whearas it loads fine in React <img>

### ISSUE THREAD

**Issue by the5ereneRebe1** [diegomura/react-pdf#1253]
**Describe the bug**
When i load an image to render it in a PDF using react-pdf `<Image>` tag i get a CORS error on my browser, while this works fine while loading the image in a React component using the `<img>` tags.

**Request Headers when loading through react-pdf `<Image>` tag that fails:**

GET /Images/Growthstage/HR/wheat_bbch_51.png HTTP/1.1
Host: x-df-basf-filestorage-masterdata-prod.s3.eu-central-1.amazonaws.com
Connection: keep-alive
Pragma: no-cache
Cache-Control: no-cache
sec-ch-ua: " Not A;Brand";v="99", "Chromium";v="90", "Google Chrome";v="90"
sec-ch-ua-mobile: ?0
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.72 Safari/537.36
Accept: */*
Origin: https://xac-dev.xarvio.com
Sec-Fetch-Site: cross-site
Sec-Fetch-Mode: cors
Sec-Fetch-Dest: empty
Referer: https://xac-dev.xarvio.com/
Accept-Encoding: gzip, deflate, br
Accept-Language: en,en-US;q=0.9

**Error message :**
Access to XMLHttpRequest at 'https://x-df-basf-filestorage-masterdata-prod.s3.eu-central-1.amazonaws.com/Images/Growthstage/LR/barley_bbch_59.png' from origin 'https://xac-dev.xarvio.com' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.

**Request Headers when loading through React component `<img>` tag:**

GET /Images/Growthstage/HR/rye_bbch_31.png HTTP/1.1
Host: x-df-basf-filestorage-masterdata-prod.s3.eu-central-1.amazonaws.com
Connection: keep-alive
Pragma: no-cache
Cache-Control: no-cache
sec-ch-ua: " Not A;Brand";v="99", "Chromium";v="90", "Google Chrome";v="90"
sec-ch-ua-mobile: ?0
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.72 Safari/537.36
Accept: image/avif,image/webp,image/apng,image/svg+xml,image/*,*/*;q=0.8
Sec-Fetch-Site: cross-site
Sec-Fetch-Mode: no-cors
Sec-Fetch-Dest: image
Referer: https://xac-dev.xarvio.com/
Accept-Encoding: gzip, deflate, br
Accept-Language: en,en-US;q=0.9

**Expected behavior:**
The image should load the same way using both the methods.

**Desktop (please complete the following information):**
 - OS: [Windows]
 - Browser [chrome]
 - React-pdf version [e.g. v1.6.13]

**Comment by vasilenka** [comment]
I've got this issue too just now
I solved it by adding the origin to the allow-origin list on my S3 config

**Comment by diegomura** [comment]
@vasilenka is right! The server should allow the origin. There's no much to do on our side if the server is not returning the img

**Comment by the5ereneRebe1** [comment]
I had just one doubt , hope someone can spare some time to explain , but how come loading the image in the `<img>` tag in a react component is not causing any CORS issue, i really want to understand how this is working.

**Comment by reinvanimschoot** [comment]
I would also really like to know and understan why an `<img>` works fine but the `Image` is giving me trouble.

**Comment by oliverterrell** [comment]
@the5ereneRebe1 @reinvanimschoot Same question here

@diegomura can we get an answer? It seems there is something you can be doing on your side given this disparity

**Comment by jeetiss** [comment]
`react-pdf` download the image binary with fetch API to embed it to pdf and `fetch` follows same-origin policy.
`react` propagate the image src to the browser and the browser downloads image without any cors check.

that's the difference

https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS

**Comment by oliverterrell** [comment]
> `react-pdf` download the image binary with fetch API to embed it to pdf and `fetch` follows same-origin policy. `react` propagate the image src to the browser and the browser downloads image without any cors check.
> 
> that's the difference
> 
> https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS

Thank you that makes perfect sense. The nature of my work is such that I can't be sending internal data to an external server. Are there any plans to mimic browser propagation in react to keep everything local?

**Comment by jeetiss** [comment]
> Are there any plans to mimic browser propagation in react to keep everything local?

No, that is impossible

**Comment by Chakkaphong9** [comment]
thank you that make me understand how cors work, in my case i am fix with #929 by add 
`<Image src={{ uri: _your_image_url_goes_here_, method: "GET", headers: { "Cache-Control": "no-cache" }, body: "" }} />`

**Comment by alexlu0917** [comment]
I faced this error as well.
Can anyone let me know how to fix it?

**Comment by Nutriadmin** [comment]
Please check #2340 for how I've solved this issue by making my server convert images to base64. This will work even when loading images from an external website that has a CORS policy you cannot control.

**Comment by nimeshmaharjan1** [comment]
> I've got this issue too just now I solved it by adding the origin to the allow-origin list on my S3 config

what did you add tho

**Comment by laboratoryyingong** [comment]
> I faced this error as well. Can anyone let me know how to fix it?

https://github.com/diegomura/react-pdf/issues/1253#issuecomment-1475861858

**Comment by dgreasi** [comment]
Check this [comment](https://github.com/diegomura/react-pdf/issues/929#issuecomment-1884612597). It might help.

**Comment by timeemit** [comment]
I had an unusual case where I was asking react-pdf to render an image from a URL that it received from a redirect.  CORS failed any requests for the redirection until I passed in a function for the src:

```javascript
<PDFImage src={() => fetch(originalURL).then(redirectionResponse => redirectionResponse.url)} ... />
```

**Comment by Fahim-ju** [comment]
> thank you that make me understand how cors work, in my case i am fix with #929 by add `<Image src={{ uri: _your_image_url_goes_here_, method: "GET", headers: { "Cache-Control": "no-cache" }, body: "" }} />`

Thanks... It works for me.
